src/engraving/rendering/dev/guitarbendlayout.cpp

```diff
@@ -0,0 +1,589 @@
+    item->mutldata()->setPath(path);
+}
+
+void GuitarBendLayout::checkConflictWithOtherBends(GuitarBendSegment* item)
+{
+    GuitarBend* thisBend = item->guitarBend();
+    Note* startNote = thisBend->startNote();
+    Chord* startChord = startNote->chord();
+
+    GuitarBend* otherBend = nullptr;
+    for (Note* note : startChord->notes()) {
+        if (note->string() >= startNote->string()) {
+            continue;
+        }
+        for (Spanner* sp : note->spannerFor()) {
+            if (sp->isGuitarBend()) {
+                otherBend = toGuitarBend(sp);
+                break;
+            }
+        }
+    }
+
+    if (!otherBend) {
+        return;
+    }
+
+    if (otherBend->ldata()->bendDigit() != thisBend->ldata()->bendDigit()) {
+        if (thisBend->type() != GuitarBendType::PRE_BEND) {
+            GuitarBendSegment::LayoutData* ldata = item->mutldata();
+            double spatium = item->spatium();
+            PointF endPointMove(spatium, spatium);
+            item->setPos2(item->pos2() + endPointMove);
+            item->mutldata()->arrow().translate(endPointMove);
+            ldata->setVertexPoint(ldata->vertexPoint() + PointF(endPointMove.x(), 0.0));
+
+            GuitarBendText* text = item->bendText();
+            text->mutldata()->setPos(text->ldata()->pos() + endPointMove + PointF(0.35 * text->width(), 0.0));
+        }
+    }
+}
+
+void GuitarBendLayout::layoutTabStaff(GuitarBendSegment* item, LayoutContext& ctx)
+{
+    GuitarBend* bend = item->guitarBend();
+    GuitarBendSegment::LayoutData* ldata = item->mutldata();
+
+    Note* startNote = bend->startNote();
+    Note* endNote = bend->endNote();
+    if (!startNote || !endNote) {
+        return;
+    }
+
+    double spatium = bend->spatium();
+    System* system = item->system();
+
+    const double horizontalIndent = 0.25 * spatium;
+    const double verticalPad = 0.25 * spatium;
+    const double distAboveTab = 1.5 * spatium * item->staff()->lineDistance(item->tick());
+    const double partialBendHeight = 1.5 * spatium;
+    const double arrowWidth = 1.0 * spatium;
+    const double arrowHeight = 1.0 * spatium;
+
+    PointF startPos;
+    PointF endPos;
+    PointF vertex;
+    PolygonF arrow;
+
+    Chord* startChord = startNote->chord();
+    PointF startNotePos = startNote->pos() + startChord->pos() + startChord->segment()->pos() + startChord->measure()->pos();
+    Chord* endChord = endNote->chord();
+    PointF endNotePos = endNote->pos() + endChord->pos() + endChord->segment()->pos() + endChord->measure()->pos();
+
+    GuitarBend* prevBend = bend->findPrecedingBend();
+    GuitarBendHoldSegment* prevHoldLine = prevBend && prevBend->holdLine() && !prevBend->holdLine()->segmentsEmpty()
+                                          ? toGuitarBendHoldSegment(prevBend->holdLine()->backSegment()) : nullptr;
+
+    GuitarBendType type = bend->type();
+    if (type != GuitarBendType::PRE_BEND) {
+        if (bend->isReleaseBend()) {

```

Comment:
Ideally, it would be great to add a few private methods to avoid a lot of nested if-else statements

Reply:
Yeah, this one came out very messy, I'll clean it up as soon as the engraving result is approved, so I can refactor it by checking against the vtests